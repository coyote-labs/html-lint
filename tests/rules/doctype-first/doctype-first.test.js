const lint = require('../../../index');
const { getFileMeta } = require('../../utils');

global.console = {
  error: jest.fn()
};

test.each([
  'rules/doctype-first/no-error.html',
  'rules/doctype-first/no-error-leading-comments.html',
  'rules/doctype-first/error-no-doctype.html',
  'rules/doctype-first/error-no-doctype-leading-comments'
])(
  'doctype-first %s',
  async(fixture) => {
    try {
      await lint([getFileMeta(fixture)]);
    } catch ({ message }) {
      expect(global.console.error).toMatchSnapshot();
    }
  },
);
