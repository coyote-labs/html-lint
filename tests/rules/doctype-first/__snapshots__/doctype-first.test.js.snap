// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`doctype-first rules/doctype-first/error-no-doctype.html 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "[101m[html-lint] 1 error in rules/doctype-first/error-no-doctype.html[49m",
    ],
    Array [
      "[90mdoctype-first: HTML files should begin with a \`doctype\`.[39m
[0m[31m[1m>[22m[39m[90m 1 | [39m[33m<[39m[33mhtml[39m[33m>[39m[33m<[39m[33m/[39m[33mhtml[39m[33m>[39m[0m
[0m [90m   | [39m[31m[1m^[22m[39m [31m[1mHTML files should begin with a \`doctype\`.[22m[39m[0m",
      "
",
    ],
    Array [
      "
",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`doctype-first rules/doctype-first/error-no-doctype-leading-comments 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "[101m[html-lint] 1 error in rules/doctype-first/error-no-doctype.html[49m",
    ],
    Array [
      "[90mdoctype-first: HTML files should begin with a \`doctype\`.[39m
[0m[31m[1m>[22m[39m[90m 1 | [39m[33m<[39m[33mhtml[39m[33m>[39m[33m<[39m[33m/[39m[33mhtml[39m[33m>[39m[0m
[0m [90m   | [39m[31m[1m^[22m[39m [31m[1mHTML files should begin with a \`doctype\`.[22m[39m[0m",
      "
",
    ],
    Array [
      "
",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
